src/item.cpp

```diff
@@ -5685,10 +5772,15 @@ void item::fill_with( item &liquid, long amount )
         }
         ammo_set( liquid.typeId(), ammo_remaining() + amount );
     } else if( !is_container_empty() ) {
-        contents.front().mod_charges( amount );
+        // make a copy of the current contents, modify it, then replace current
+        // contents with copy
```

Comment:
I can see *what* the code does, but *why* is it done that way. Why is the item in the container not changed directly?

Reply:
It seemed good at the time?  I was going off the `else` below that makes a copy of the liquid and assumed it had a good reason but now looking closer at it I understand why it's making a copy of the passed in liquid and why I don't need to do that.